examples/heterogeneous-node-classification-with-hashgnn.ipynb

```diff
@@ -0,0 +1,534 @@
+    "\n",
+    "Please see the [HashGNN documentation](https://neo4j.com/docs/graph-data-science/current/machine-learning/node-embeddings/hashgnn/) for more on this algorithm."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5e254efe",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Add a HashGNN node property step to the pipeline\n",
+    "_ = pipe.addNodeProperty(\n",
+    "    \"beta.hashgnn\",\n",
+    "    mutateProperty=\"embedding\",\n",
+    "    iterations=4,\n",
+    "    heterogeneous=True,\n",
+    "    embeddingDensity=512,\n",
+    "    neighborInfluence=0.8,\n",
+    "    featureProperties=[\"plot_keywords\"],\n",
+    "    randomSeed=41,\n",
+    "    contextNodeLabels=G.node_labels(),\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "30ff6713",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Set the embeddings vectors produced by HashGNN as feature input to our ML algorithm\n",
+    "_ = pipe.selectFeatures(\"embedding\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6f4f2bc1",
+   "metadata": {},
+   "source": [
+    "## Setting up autotuning\n",
+    "\n",
+    "It is time to set up the [ML algorithms](https://neo4j.com/docs/graph-data-science/current/machine-learning/training-methods/) for the training part of the pipeline.\n",
+    "\n",
+    "In this example we will add logistic regression and random forest algorithms as candidates for the final model.\n",
+    "Each candidate will be evaluated by the pipeline, and the best one, according to our specified metric, will be chosen.\n",
+    "\n",
+    "It is hard to know how much regularization we need so as not to overfit our models on the training dataset, and for this reason we will use the autotuning capabilities of GDS to help us out.\n",
+    "The autotuning algorithm will try out several values for the regularization parameters `penalty` (of logistic regression) and `minSplitSize` (of random forest) and choose the best ones it finds.\n",
+    "\n",
+    "Please see the GDS manual to learn more about [autotuning](https://neo4j.com/docs/graph-data-science/current/machine-learning/auto-tuning/), [logistic regression](https://neo4j.com/docs/graph-data-science/current/machine-learning/training-methods/logistic-regression/) and [random forest](https://neo4j.com/docs/graph-data-science/current/machine-learning/training-methods/random-forest/)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8bccd326",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Add logistic regression as a candidate ML algorithm for the training\n",
+    "# Provide an interval for the `penalty` parameter to enable autotuning for it\n",
+    "_ = pipe.addLogisticRegression(penalty=(0.1, 1.0), maxEpochs=1000, patience=5, tolerance=0.0001, learningRate=0.01)"
```

Comment:
this penalty interval is rather small. in sklearn the default is (1e-4, 1e4) . is the smaller interval a result of manual tuning?

Reply:
I think I just copied this from the docs or something haha. Seems to work well enough, but yeah it's a rather specific range :man_shrugging: